waiter/src/waiter/handler.clj

```diff
@@ -493,25 +492,34 @@
 
 (defn get-router-state
   "Outputs the state of the router as json."
-  [state-chan scheduler-chan router-metrics-state-fn kv-store leader?-fn scheduler request]
+  [state-chan scheduler-chan router-metrics-state-fn kv-store leader?-fn request]
   (try
-    (let [timeout-ms 30000
+    (let [embed-param (-> request
+                          ring-params/params-request
+                          :params
+                          (get "embed"))
+          embed-pred (cond
+                       (string? embed-param) #{embed-param}
+                       (sequential? embed-param) (set embed-param)
+                       :else (constantly true))
```

Comment:
How do I just get state?

Reply:
state is always included by default. So passing `embed=state` will achieve that.

Reply:
While `embed=state` works given the implementation, it's an awkward API. We don't "embed" state within state.  

We have an opportunity to clean things up, making `/state`, by default, return just state, and embed options add to the output.

Another opportunity is to create dedicated endpoints for the specific states, e.g.
`/state/scheduler`
`/state/statsd`
etc.